### PR TITLE
NAB 2024 demo code changes ported

### PR DESCRIPTION
**Pull request by Mionsz**
NAB 2024 demo code changes ported and described accordingly in separate commit messages. 

[b460cd4 Extended debug information](https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/117/commits/b460cd4e74e666dfe85ab9e9ef42852fbfe25d3b) 
- Extended debug information added
- Deprecated migration for IP address structures
- IMTL flag `MTL_FLAG_BIND_NUMA` added. This, in multi-NUMA environments forces IMTL to try assigning used CPU cores correctly, in regards to NIC interface hardware NUMA number, basically allowing speed boost. 

[b5e5d69 Extended MCM structures](https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/117/commits/b5e5d69cb6dd7f32abbab7fb80201d790f1dc3aa) 
- Added PIX_FMT_YUV422P_10BIT_LE pixel format
- Added payload type, IMTL mask and IMTL pacing parameters
- Made all above configurable.

[307a628 Timestamp extraction from frame](https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/117/commits/307a628fc4502874d99b8bb6cb69f92a91ad5ba7) 

Timestamp extraction, using IMTL timestamp parameter, from received frame added.

[4ba6854 Extended rx_st20p_query_ext_frame method](https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/117/commits/4ba685435dd597d14f1c0007fbb8dfe5c8e7c509) 

Changes to default behavior, regarding used FMT formats made. Implementation for external frame added.
- Extended rx_st20p_query_ext_frame method to included planes based on frame FMT
- Changed default transport FMT to ST20_FMT_YUV_422_10BIT
- Changed default input/output FMT to ST_FRAME_FMT_YUV422PLANAR10LE

[27e6eec Receiver and Sender SDK updated](https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/117/commits/27e6eecddfa95e8f62f4134563f9c6ebf930333b) 
Updated SDK sample applications to include changes made in previous commits in this PR:
- Receiver and Sender apps updated
- Added more configurable options

[194132b Update ubuntu-build.yml](https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/117/commits/194132b7c3cd40e0344b5ebf1d4fef72ba9680cd) 

Updated workflow for `ubuntu-build.yml` to fix the Ci/Cd process.
Added missing apt-get install components.
Minor issues fixed.

### CODE REVIEW

File: media-proxy/src/mtl.c [hunk 1]
```diff
@@ -1270,13 +1270,11 @@ int tx_st20p_shm_init(tx_session_context_t* tx_ctx, memif_ops_t* memif_ops)
     }
 
     /* Set application name */
-    strncpy(tx_ctx->memif_socket_args.app_name, memif_ops->app_name,
-        sizeof(tx_ctx->memif_socket_args.app_name) - 1);
+    strncpy(tx_ctx->memif_socket_args.app_name, memif_ops->app_name, sizeof(tx_ctx->memif_socket_args.app_name) - 1);
```
Comment (ko80) [inline media-proxy/src/mtl.c:1297]: I suppose these `strncpy` cases could be replaced with `strlcpy` which has a better implementation https://linux.die.net/man/3/strlcpy. This is for the future refactoring. I'm not requesting to update this PR

File: media-proxy/src/proxy_context.cc [hunk 2]
```diff
@@ -758,7 +776,7 @@ int ProxyContext::RxStart(const mcm_conn_param* request)
         ParseStInitParam(request, &st_param);
 
         /* set default parameters */
-        // st_param.flags = MTL_FLAG_BIND_NUMA;
+        st_param.flags = MTL_FLAG_BIND_NUMA;
```
Comment (ko80) [inline media-proxy/src/proxy_context.cc:802]: This seems not related to the PR scope. Could you elaborate more? Why you enable this flag?

Comment (Mionsz) [reply media-proxy/src/proxy_context.cc:802]: In multi-numa environments this flags try to assign cores correctly in regards to used interface hardware numa number, basically allowing speed boost. This was mandatory for the presentation.

File: media-proxy/src/mtl.c (hunk 1 above)
Comment (Mionsz) [reply media-proxy/src/mtl.c:1297]: I did not know that, thanks for the notice. There are lots of issues that need to be resolved and string copy is one of them.

File: media-proxy/src/proxy_context.cc (hunk 2 above)
Comment (ko80) [reply media-proxy/src/proxy_context.cc:802]: Maybe you could amend the commit message to cover that change? But it's up to you.

Comment (Mionsz) [reply media-proxy/src/proxy_context.cc:802]: I can add this to the description of PR as this flag was added in first commit that was done.
